### PR TITLE
Order checkout stock

### PR DESCRIPTION
**Pull request by RUKAIYIN**
On receiving `OrdersTopics.ORDER_CHECKED_OUT`, the Stock consumer will check if all the stock items in the order are available for subtracting and emit `OrdersTopics.ORDER_PROCESSED_IN_STOCK_SUCC` if true and `OrdersTopics.ORDER_PROCESSED_IN_STOCK_FAILED` otherwise.

### CODE REVIEW

File: common/src/main/java/nl/tudelft/wdm/group1/common/OrdersTopics.java [hunk 1]
```diff
@@ -6,8 +6,11 @@
     public static final String ORDER_DELETED = "orderDeleted";
     public static final String ORDER_ITEM_ADDED = "orderItemAdded";
     public static final String ORDER_ITEM_DELETED = "orderItemDeleted";
+    public static final String ORDER_PROCESSED_IN_STOCK_SUCC = "orderProcessedInStockSuccessful";
```
Comment (yorickdevries) [inline common/src/main/java/nl/tudelft/wdm/group1/common/OrdersTopics.java]: I would opt for ORDER_PROCESSED_IN_STOCK_SUCCESSFUL

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java [hunk 2]
```diff
@@ -24,4 +31,56 @@ public void consume(final StockItem stockItem) {
...
+            }
+
+            try {
+                stockItem.subtractStock(1);
+            } catch (InsufficientStockException | InvalidStockChangeException e) {
+                // we assume the subtraction should not cause any exception
+                throw e;
+            }
+
+            totalPrice += stockItem.getPrice() * 1;
```
Comment (yorickdevries) [inline stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java]: Why is there a ```* 1``` here?

Comment (RUKAIYIN) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java]: The amount of the stock item

File: common/src/main/java/nl/tudelft/wdm/group1/common/OrdersTopics.java (hunk 1 above)
Comment (RUKAIYIN) [reply common/src/main/java/nl/tudelft/wdm/group1/common/OrdersTopics.java]: ok

File: stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java (hunk 2 above)
Comment (yorickdevries) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java]: But thats always 1, why should there be a ```* 1``` then?

Comment (gandreadis) [reply stock/src/main/java/nl/tudelft/wdm/group1/stock/events/Consumer.java]: Done :)

**Pull request merged**